src/lib/support/BitFlags.h

```diff
@@ -87,6 +88,40 @@ class BitFlags
         return *this;
     }
 
+    /**
+     * SetField to value via a mask shifting.
+     *
+     * @param mask      Typed flag(s) to used as mask, flag(s) must be contiguous. Any flags not in @a v are unaffected.
+     * @param value     New value to apply at the field/mask location.
+     */
+    constexpr BitFlags & SetField(FlagsEnum mask, IntegerType value)
```

Comment:
This is unclear to me: please add examples what this is intended to do.

How does this ensure that all bits set are valid? I believe all other methods on this class are type safe.